### PR TITLE
Move exception to setter for UsePollingFileWatcher

### PR DESCRIPTION
**Pull request by maryamariyan**
Getters on properties shouldn't throw exceptions.

 - Moves exception to setter for UsePollingFileWatcher

Addresses #561

cc: @anurse

### CODE REVIEW

File: src/FileProviders/Physical/src/PhysicalFileProvider.cs [hunk 1]
```diff
@@ -84,21 +84,19 @@ public PhysicalFileProvider(string root, ExclusionFilters filters)
...
+            set
             {
                 if (_fileWatcher != null)
                 {
                     throw new InvalidOperationException($"Cannot modify {nameof(UsePollingFileWatcher)} once file watcher has been initialized.");
                 }
-
                 if (_usePollingFileWatcher == null)
                 {
                     ReadPollingEnvironmentVariables();
```
Comment (anurse) [inline src/FileProviders/Physical/src/PhysicalFileProvider.cs]: I think we need to keep this `if` clause in the `get` accessor. It's just the exception that needs to move.

File: src/FileProviders/Physical/test/PhysicalFileProviderTests.cs [hunk 2]
```diff
@@ -1378,6 +1378,63 @@ public async Task WildCardToken_RaisesEventsWhenFileSystemWatcherDoesNotFire()
...
+                            // Act / Assert
+                            Assert.Throws<InvalidOperationException>(() => { provider.UsePollingFileWatcher = true; });
+                        }
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public void UsePollingFileWatcher_FileWatcherNotNull_GetterEfzUp()
```
Comment (BrennanConroy) [inline src/FileProviders/Physical/test/PhysicalFileProviderTests.cs]: `EfzUp` ??

Comment (maryamariyan) [reply src/FileProviders/Physical/test/PhysicalFileProviderTests.cs]: Please ignore. I thought I removed that.

File: src/FileProviders/Physical/src/PhysicalFileProvider.cs [hunk 3]
```diff
@@ -86,19 +86,20 @@ public bool UsePollingFileWatcher
         {
             get
             {
-                if (_fileWatcher != null)
-                {
-                    throw new InvalidOperationException($"Cannot modify {nameof(UsePollingFileWatcher)} once file watcher has been initialized.");
-                }
-
                 if (_usePollingFileWatcher == null)
```
Comment (BrennanConroy) [inline src/FileProviders/Physical/src/PhysicalFileProvider.cs]: I think we also want to keep:
```csharp
if (_fileWatcher != null)
{
    return false;
}
```

File: src/FileProviders/Physical/src/PhysicalFileProvider.cs [hunk 4]
```diff
@@ -86,19 +86,20 @@ public bool UsePollingFileWatcher
...
                 {
                     ReadPollingEnvironmentVariables();
                 }
-
-                return _usePollingFileWatcher.Value;
+                return _usePollingFileWatcher ?? false;;
```
Comment (BrennanConroy) [inline src/FileProviders/Physical/src/PhysicalFileProvider.cs]:
```suggestion
                return _usePollingFileWatcher ?? false;
```

File: src/FileProviders/Physical/src/PhysicalFileProvider.cs (hunk 3 above)
Comment (BrennanConroy) [reply src/FileProviders/Physical/src/PhysicalFileProvider.cs]: I think the test you removed was good and will verify this check is here

**Comment by maryamariyan** [comment]
@anurse is this PR ok to merge? I don't have permission to merge myself.

**Pull request merged**